plugin/pkg/scheduler/factory/plugins.go

```diff
@@ -118,6 +118,9 @@ func RegisterMandatoryFitPredicate(name string, predicate algorithm.FitPredicate
 func RegisterFitPredicateFactory(name string, predicateFactory FitPredicateFactory) string {
 	schedulerFactoryMutex.Lock()
 	defer schedulerFactoryMutex.Unlock()
+	if predicateFactory == nil {
```

Comment:
What would happen if missing this nil pointer checking?

I think we should not introduce unnecessary changes if everything works well.

Reply:
Also I think the checking of `predicateFactory` should be done in the upper calling method.

Reply:
+1 to @dixudx. The addition of this check is not necessary in this PR.